yb-voyager/cmd/common.go

```diff
@@ -393,6 +393,7 @@ func CreateMigrationProjectIfNotExists(dbType string, exportDir string) {
 	setSourceDbType(dbType, projectDirPath)
 
 	schemaObjectList := utils.GetSchemaObjectList(dbType)
+	utils.ApplyExportSchemaObjectListFilter(&schemaObjectList, source.ExportObjects)
```

Comment:
Assuming the user has passed correct object types for the current `dbType`, this statement is equivalent to `strings.Split(source.ExportObjectTypes)`.

And all the calls to the `ApplyExportSchemaObjectListFilter()` are preceded by the call to `utils.GetSchemaObjectList()`.

How about the following changes:
- Collect the user input for the flag in `source.StrExportObjectTypeList`.
- Towards the very beginning of the `export schema`, validate the `StrExportObjectTypeList` and prepare the final `source.ExportObjectTypeList` (which is a slice of strings). The object-type names in the list are converted to all caps. The validation must ensure that the object-type provided by the user is indeed applicable for the `dbType` in consideration.
- Use the `source.ExportObjectTypeList` directly throughout the code.